include/biosHandler.hpp

```diff
@@ -4,16 +4,81 @@
 
 namespace vpd
 {
+
+/**
+ * @brief Interface class for BIOS handling.
+ *
+ * The class layouts virtual methods required to be implemented by any concrete
+ * class intended to use feature provided via BIOS handler class.
+ */
+class BiosHandlerInterface
+{
+  public:
+    /**
+     * @brief API to back up or restore BIOS attributes.
+     *
+     * Concrete class should implement the API and read the backed up data from
+     * its designated location and take a call if it should be backed up or
+     * restored.
+     */
+    virtual void backUpOrRestoreBiosAttributes() = 0;
+
+    /**
+     * @brief API to be triggered on BIOS attribute change.
+     *
+     * Concrete class should implement the API to capture any change in the BIOS
+     * attribues and in the implementation needs to filter out the attribute it
+     * is interested.
+     *
+     * @param[in] i_msg - The callback message.
+     */
+    virtual void biosAttribsCallback(sdbusplus::message_t& i_msg) = 0;
+};
+
+/**
+ * @brief IBM specifc BIOS handler class.
+ */
+class IbmBiosHandler : public BiosHandlerInterface
+{
+  public:
+    /**
+     * @brief API to back up or restore BIOS attributes.
+     *
+     * The API will read the backed up data from the VPD keyword and based on
+     * its value, either backs up or restores the data.
+     */
+    virtual void backUpOrRestoreBiosAttributes();
+
+    /**
+     * @brief API to be triggered on BIOS attribute change.
+     *
+     * The API listens for specific BIOS attributes and backs them up to VPD
+     * keyword.
+     *
+     * @param[in] i_msg - The callback message.
+     */
+    virtual void biosAttribsCallback(sdbusplus::message_t& i_msg);
+};
+
 /**
- * @brief A class to operate upon and back up some of the BIOS attributes.
+ * @brief A class to operate upon BIOS attributes.
+ *
+ * The class along with specific BIOS handler class(es), provides a feature
+ * where specific BIOS attributes identified by the concrete specifc class can
+ * be listened for any change and can be backed up to a desired location or
+ * restored back to the BIOS table.
+ *
+ * To use the feature, "BiosHandlerInterface" should be implemented by a
+ * concrete class and the same should be used to instantiate BiosHandler.
  *
- * The class initiates call backs to listen to PLDM service and changes in some
- * of the BIOS attributes.
- * On a factory reset, BIOS attributes are initialized by PLDM. this class waits
- * until PLDM has grabbed a bus name before attempting any syncs.
+ * This class initiates call backs to listen to PLDM service as it is being used
```

Comment:
PLDM service may perform other operations apart from BIOS attributes handling right ?
should we rephrase this statement ?

ex: "listen to PLDM service as it handles reading/writing of BIOS attributes." ?